pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -315,20 +325,33 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 			return reconcile.Result{}, err
 		}
 	}
+	// Default fields on the LogCollector instance if needed.
+	if r.enterpriseCRDExists {
```

Comment:
I think this should be based on the Variant from the Installation resource instead of the CRDs.